File: week6programs/sentimental-mario-more/mario.py
from cs50 import get_int


def main():
    height = get_height()
    for i in range(1, height + 1):
        print(" " * (height - i) + "#" * i, end="")  # Left side of pyramid
        print("  ", end="")  # Spaces inbetween
        print("#" * i, end="")  # Right side of pyramid
        print()  # Creates new line


def get_height():
    while True:
        height = get_int("Height: ")
        if height > 0 and height < 9:
            return height


if __name__ == "__main__":
    main()

File: week1programs/pset1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Get the height of the blocks from the user
    int size;
    do
    {
        size = get_int("Height: ");
    }
    while (size < 1 || size > 8);

    // Print out the blocks
    int initial_spaces = size - 1;
    for (int i = 0; i < size; i++) // This loop keeps track of the current row
    {
        for (int j = 0; j < initial_spaces; j++) // Keeps track of the number of spaces intially
        {
            printf(" ");
        }

        for (int k = 0; k < i + 1; k++) // Prints out the number of hashes on the left side
        {
            printf("#");
        }

        printf("  "); // Prints the two spaces between the blocks

        for (int l = 0; l < i + 1; l++) // Prints out the number of hashes on the right side
        {
            printf("#");
        }

        printf("\n"); // Creates new line to continue to the next row
        initial_spaces--; // Keeps track of the number of spaces to be printed out
    }

}